File: Week 6/sentimental-readability/readability.py
import cs50


def main():

    # A list of grades
    grades = ["Before Grade 1", "Grade 1", "Grade 2", "Grade 3", "Grade 4", "Grade 5", "Grade 6", "Grade 7",
              "Grade 8", "Grade 9", "Grade 10", "Grade 11", "Grade 12", "Grade 13", "Grade 14", "Grade 15", "Grade 16+"]

    text = cs50.get_string("Enter Text: ")

    # Pass the input text into a function, that returns multiple values
    letters, words, sentences = count_all(text)

    # Pass the data into the function, running the index formula, thereby outputting a grade from our list
    result = index_formula(letters, words, sentences, grades)

    print(result)

# Loops through each character, checking if letter, word, or sentence. Word starts at 1, because at the end of a sentence there is no space.


def count_all(text):
    letters = 0
    words = 1
    sentences = 0

    for c in text:
        if c.isalpha():
            letters += 1
            continue
        if c.isspace():
            words += 1
            continue
        if c == ".":
            sentences += 1
            continue
        if c == "!":
            sentences += 1
            continue
        if c == "?":
            sentences += 1

    return letters, words, sentences

# Based on the formula results, loops through our list until it finds a match


def index_formula(letters, words, sentences, grades):
    average_letters = (letters / words) * 100
    average_sentences = (sentences / words) * 100

    result = 0.0588 * average_letters - 0.296 * average_sentences - 15.8
    result = round(result)

    if result < 1:
        return grades[0]
    elif result > 16:
        return grades[16]

    grading = 2

    for i in range(2, 17, 1):
        if result == grading:
            return grades[grading]
        grading += 1


main()

File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);
string index(int letters, int words, int sentences, string grades[]);


int main(void)
{
    string grades[] = {"Before Grade 1", "Grade 1", "Grade 2", "Grade 3", "Grade 4", "Grade 5", "Grade 6", "Grade 7", "Grade 8", "Grade 9", "Grade 10", "Grade 11", "Grade 12", "Grade 13", "Grade 14", "Grade 15", "Grade 16+"};
    string word = get_string("Text: ");
    int letters = count_letters(word);
    int words = count_words(word);
    int sentences = count_sentences(word);
    string result = index(letters, words, sentences, grades);cd

    // Prints grade
    printf("%s\n", result);
}









// Counts alphabetic letters
int count_letters(string word)
{
    int letters = 0;
    int inputlength = strlen(word);

    for (int i = 0; i < inputlength; i++)
    {
        char c = word[i];
        if (isalpha(c))
        {
            letters++;
        }
    }
    return letters;
}

// counts words based on spaces + 1
int count_words(string word)
{
    int words = 0;
    int inputlength = strlen(word);

    for (int i = 0; i < inputlength; i++)
    {
        char s = word[i];
        if (isspace(s))
        {
            words++;
        }
    }
    return words += 1;
}

// Counts sentences
int count_sentences(string word)
{
    int sentences = 0;
    int inputlength = strlen(word);

    for (int i = 0; i < inputlength; i++)
    {
        char k = word[i];
        if ((k == '.') || (k == '?') || (k == '!'))
        {
            sentences++;
        }
    }
    return sentences;
}


// Grade coleman formula index that returns the grade
string index(int letters, int words, int sentences, string grades[])
{
    string output = "aaaaaaaaaaaaaa";
    float i = 0.00;
    float l = (((float) letters / (float) words) * 100);
    float s = (((float) sentences / (float) words) * 100);
    i = 0.0588 * l - 0.296 * s - 15.8;
    i = round(i);

    if (i < 1)
    {
        output = grades[0];
    }
    if (i > 16)
    {
        output = grades[16];
    }

    for (int j = 1; j < 17; j++)
    {
        if (i == j)
        {
            output = grades[j];
        }
    }
    return output;
}